src/parser/core/Combatant.ts

```diff
@@ -130,7 +130,7 @@ class Combatant extends Entity {
   }
 
   _getTalent(row: number) {
-    return this._talentsByRow[row];
+    return [...this._talentsByRow][row];
```

Comment:
this spread / array construction likely eliminates all of the gain from using a set for the `.has(...)`.

Reply:
This is one thing I feared but this method is almost never called 
If I did my checking correctly the only places that directly call_getTalent is the `lvFooTalent` methods which those are called a grand total of 5 times throughout the code base specifically in spec analyzers.
If anything this set of code `_getTalent` -> `getlvFooTalent` should probably just be deleted all in all

Looking forwards to dragon flight (which is probably not the correct call) since the tree is now not deterministic as before this code would most likely die there.